### PR TITLE
[feature/APPC-1162] Merge APPC and APPC-C

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

 This PR aims to merge both payment methods in a single one. After this PR the user is presenter with a appc/appc-c payment method and then navigates to another view showing both in separate.

**Database changed?**

 No

**Where should the reviewer start?**

- MergedAppcoinsFragment.kt
- PaymentMethodsFragment.java
- merged_appcoins_layout.xml

**How should this be manually tested?**

-On the PaymentMethodsView:
--Check if button is set to Next when the merge method is selected and buy or donate in the other cases;
--Check if when the users has no appc or appc-c if the method shows as disable
-On the MergedAppcoinsView:
-Go with only appc and then only appc-c and check if one of the method is disable (both portrait and landscape
-Go with both methods enabled and check if selection is working correctly and check if the layout changes correctly on landscape (bonus)
-Make a payment with each of the methods at least once
-Make a payment with one of the methods on donation, oil and freeze
-Press back both on button and on device and check if it navigates back correctly 
-Press back on device immediately after pressing back on MergedAppcoinsView

**What are the relevant tickets?**

https://app.zeplin.io/project/5ba253dfbf8fc95444761ac4/screen/5d175198489aac5700a262c2
https://aptoide.atlassian.net/browse/APPC-1162

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/ui/iab/IabActivity.java [hunk 1]
```diff
@@ -184,10 +189,18 @@ public void showLocalPayment(String domain, String skuId, String originalAmount,
   }
 
   @Override public void showPaymentMethodsView() {
+    showPaymentMethodsView(PaymentMethodsView.SelectedPaymentMethod.CREDIT_CARD);
```
Comment (joaopmgouveia) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/IabActivity.java]: Is this abstraction necessary?

File: app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java [hunk 2]
```diff
@@ -225,6 +227,64 @@ private Payment mapToBdsPayment(Payment transaction, Purchase purchase) {
         .map(this::swapDisabledPositions);
   }
 
+  public Single<List<PaymentMethod>> getAppcoinsPaymentMethods(TransactionBuilder transaction,
```
Comment (joaopmgouveia) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java]: This method is not being used

File: app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    val packageManager = context!!.packageManager
+    val packageInfo = packageManager.getApplicationInfo(appPackage, 0)
+    return packageManager.getApplicationLabel(packageInfo)
+  }
+
+  private fun setBackListener(view: View) {
+    iabView.disableBack()
+    view.isFocusableInTouchMode = true
+    view.requestFocus()
+    view.setOnKeyListener { view1, keyCode, keyEvent ->
```
Comment (joaopmgouveia) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt]: Variable "view1" is not used

File: app/src/main/res/layout/payment_methods_layout.xml [hunk 4]
```diff
@@ -22,6 +22,7 @@
         android:layout_height="80dp"
```
Comment (joaopmgouveia) [inline app/src/main/res/layout/payment_methods_layout.xml:22]: The "no_bonus_msg" TextView should be deleted

**Review by joaopmgouveia — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+    if (appcEnabled) {
+      appcoins_radio.setOnClickListener { appcoins_radio_button.isChecked = true }
+      appcoins_radio_button.setOnCheckedChangeListener { _, checked ->
+        if (checked) paymentSelectionSubject?.onNext(APPC)
+        credits_radio_button.isChecked = !checked
+      }
+      appcoins_radio_button.isEnabled = true
+    } else {
+      appcoins_radio.message.text = getString(R.string.purchase_appcoins_noavailable_body)
+      appcoins_radio.title.setTextColor(resources.getColor(R.color.btn_disable_snd_color))
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt]: `getColor(@ColorRes int id)` is deprecated.
We should use `ContextCompat.getColor(Context context, @ColorRes int id)` instead

File: app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+    super.onCreate(savedInstanceState)
+    paymentSelectionSubject = PublishSubject.create()
+    onBackPressSubject = PublishSubject.create()
+    mergedAppcoinsPresenter = MergedAppcoinsPresenter(this, CompositeDisposable())
+  }
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    if (context !is IabView) {
+      throw IllegalStateException("Regular buy fragment must be attached to IAB activity")
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt]: This should refer to the correct fragment name.
Plus: change all ocurrences of "Regular buy fragment" to the correct one 😄

File: app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      CREDITS -> view.navigateToCreditsPayment()
+      else -> Log.w(TAG, "No appcoins payment method selected")
+    }
+  }
+
+  private fun handleSelection(selection: String) {
+    when (selection) {
+      APPC -> view.showBonus()
+      CREDITS -> view.hideBonus()
+      else -> Log.w(Companion.TAG, "Error creating PublishSubject")
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsPresenter.kt]: Usage of Companion.TAG is redudant and can be replace with only "TAG"

File: app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsView.kt [hunk 8]
```diff
@@ -0,0 +1,26 @@
+package com.asfoundation.wallet.ui.iab
+
+import io.reactivex.Observable
+
+interface MergedAppcoinsView {
+
+  fun showError(errorMessage: Int)
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsView.kt]: we should use Android annotation such as @IdRes as ofter as possible

File: app/src/main/res/layout-land/appcoins_radio_button.xml [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toEndOf="@id/icon"
+      app:layout_constraintTop_toTopOf="parent"
+      >
+    <TextView
+        android:id="@+id/title"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:fontFamily="sans-serif"
```
Comment (carvalho2707) [inline app/src/main/res/layout-land/appcoins_radio_button.xml]: Shouldn't you use the existing Robot regular style?

**Review by carvalho2707 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsView.kt (hunk 8 above)
Comment (JDSM01) [reply app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsView.kt]: *often

Comment (JDSM01) [reply app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsView.kt]: *StringRes was used instead

**Review by joaopmgouveia — APPROVED** [review]

**Review by carvalho2707 — APPROVED** [review]

**Pull request merged**